src/lem/eval.rs

```diff
@@ -1,60 +1,99 @@
-use crate::lem;
+use crate::func;
 
-use super::LEM;
-use anyhow::Result;
+use super::Func;
 
-/// Lurk's step function encoded as a LEM
+/// Lurk's step function
 #[allow(dead_code)]
-pub(crate) fn step() -> Result<LEM> {
-    lem!(expr_in env_in cont_in {
+pub(crate) fn eval_step() -> Func {
+    let reduce = reduce();
+    let apply_cont = apply_cont();
+    let make_thunk = make_thunk();
+
+    func!((expr_in, env_in, cont_in): 3 => {
+        let (expr, env, cont, ctrl) = reduce(expr_in, env_in, cont_in);
+        let (expr, env, cont, ctrl) = apply_cont(expr, env, cont, ctrl);
+        let (expr, env, cont, ctrl) = make_thunk(expr, env, cont, ctrl);
+        return (expr, env, cont)
+    })
+    .unwrap()
```

Comment:
I'm not so sure about the need to `unwrap` everytime we call the macro. Looks a bit annoying, but if you have a good reason for that then it's fine.